lifecycle/src/lifecycle_service_client.cpp

```diff
@@ -321,6 +330,10 @@ int main(int argc, char ** argv)
   std::shared_future<void> script = std::async(
     std::launch::async,
     std::bind(callee_script, lc_client));
+  auto wake_exec = std::async(
+    std::launch::async,
+    std::bind(wake_executor, script, std::ref(exe)));
```

Comment:
I would say this is unnecessarily complicated. I would instead move the `callee_script` into a one-off timer with zero duration.

Or better yet, run the executor's spin in a separate thread (use `std::async` if you like) and do the `callee_script` here in the main thread, then when that function is done cancel the executor and "join" the thread it was using to spin in.